### PR TITLE
feat: add experimental tproxy iptables generation

### PR DESCRIPTION
**Pull request by bartsmykla**
### Summary

This PR introduces experimental usage of new transparent proxy iptable rules generation, behind `--experimental-transparent-proxy-engine` flag of `kumactl install transparent-proxy` commannd

### Full changelog

n/a

### Issues resolved

n/a

### Documentation

n/a

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [x] Manual testing on Universal
- [x] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

**Review by lahabana — COMMENTED** [review]
This seems fine but there's no helm chart update. We should favour doing changes to helm charts first especially now that: #4086 is merged and you can use --set everywhere,

File: pkg/transparentproxy/transparentproxy_experimental.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	if tpConfig.DryRun {
+		return builder.BuildIPTables(cfg)
+	}
+
+	return builder.RestoreIPTables(cfg)
+}
+
+func (tp *ExperimentalTransparentProxy) Cleanup(dryRun, verbose bool) (string, error) {
+	// TODO implement me
+	panic("implement me")
```
Comment (lobkovilya) [inline pkg/transparentproxy/transparentproxy_experimental.go:116]: I think it should be `panic("not implemented")`, btw is it okay that Cleanup is not implemented?

Comment (lahabana) [reply pkg/transparentproxy/transparentproxy_experimental.go:116]: I think as this is experimental it's ok

**Review by lobkovilya — APPROVED** [review]

File: pkg/transparentproxy/transparentproxy_experimental.go (hunk 1 above)
Comment (bartsmykla) [reply pkg/transparentproxy/transparentproxy_experimental.go:116]: for this version yes, especially that currently cleanup is not working as it should be

**Comment by bartsmykla** [comment]
> This seems fine but there's no helm chart update. We should favour doing changes to helm charts first especially now that: #4086 is merged and you can use --set everywhere,

We currently are not configuring init containers at all using helm chart, which should be probably changed, but I would say we can do it as a followup

**Pull request merged**